From: Jeff Skirvin <jeffrey.d.skirvin@intel.com>
Date: Mon, 20 Jun 2011 14:09:31 -0700
Subject: [PATCH 240/273] isci: Disable link layer hang detection
Git-commit: 9b917987fd16d0687afe550a02f68099419f5d43
References: FATE#311808,bnc#709528
Patch-Mainline: 3.0

Some targets exceed the hang detect timer.  Use the OS timeout to
catch hung tasks.

Signed-off-by: Jeff Skirvin <jeffrey.d.skirvin@intel.com>
Signed-off-by: Dan Williams <dan.j.williams@intel.com>
Signed-off-by: Hannes Reinecke <hare@suse.de>
---
 drivers/scsi/isci/phy.c |    7 ++-----
 1 files changed, 2 insertions(+), 5 deletions(-)

diff --git a/drivers/scsi/isci/phy.c b/drivers/scsi/isci/phy.c
index 784c9a7..93a401d 100644
--- a/drivers/scsi/isci/phy.c
+++ b/drivers/scsi/isci/phy.c
@@ -242,11 +242,8 @@ scic_sds_phy_link_layer_initialization(struct scic_sds_phy *sci_phy,
 			&sci_phy->link_layer_registers->maximum_arbitration_wait_timer_timeout);
 	}
 
-	/*
-	 * Set the link layer hang detection to 500ms (0x1F4) from its default
-	 * value of 128ms.  Max value is 511 ms.
-	 */
-	writel(0x1F4, &sci_phy->link_layer_registers->link_layer_hang_detection_timeout);
+	/* Disable link layer hang detection, rely on the OS timeout for I/O timeouts. */
+	writel(0, &sci_phy->link_layer_registers->link_layer_hang_detection_timeout);
 
 	/* We can exit the initial state to the stopped state */
 	sci_change_state(&sci_phy->sm, SCI_PHY_STOPPED);
-- 
1.6.0.2

